target/linux/pistachio/base-files/etc/diag.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+#
+# Copyright (C) 2017 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+. /lib/functions/leds.sh
+. /lib/pistachio.sh
+
+status_led="marduk:red:heartbeat"
+
+set_state() {
+
+	case "$1" in
+	preinit)
+		status_led_blink_preinit
+		;;
+	failsafe)
+		status_led_blink_failsafe
+		;;
+	preinit_regular)
+		status_led_blink_preinit_regular
+		;;
+	upgrade)
+		status_led_blink_slow
+		;;
+	done)
+		status_led_set_heartbeat
```

Comment:
@pepe2k I believe `status_led_set_on` should just be `status_led_on`, looks like a bug in your staging tree?